Makefile

```diff
@@ -35,21 +55,24 @@ clean:
 man: cowsay.1
 
 cowsay.1: cowsay.1.adoc
-	a2x --format manpage ./cowsay.1.adoc
+	$(A2X) --format manpage ./cowsay.1.adoc
 
-install: cowsay.1
-	$(INSTALL) -d $(DESTDIR)$(prefix)
-	$(INSTALL) -d $(DESTDIR)$(bindir)
+install: man
+	$(INSTALL_DIRS) $(DESTDIR)$(etccowpathddir)
+	$(INSTALL_DIRS) $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) cowsay $(DESTDIR)$(bindir)/cowsay
-	$(LN) -s cowsay $(DESTDIR)$(bindir)/cowthink
-	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
+	rm -f $(DESTDIR)$(bindir)/cowthink
+	$(LN_S) cowsay $(DESTDIR)$(bindir)/cowthink
+	$(INSTALL_DIRS) $(DESTDIR)$(mandir)/man1
 	$(INSTALL_DATA) cowsay.1 $(DESTDIR)$(mandir)/man1
-	$(INSTALL_DATA) cowthink.1 $(DESTDIR)$(mandir)/man1
-	$(INSTALL) -d $(DESTDIR)$(datadir)/cowsay
-	cp -R share/cows $(DESTDIR)$(datadir)/cowsay
-	$(INSTALL) -d $(DESTDIR)$(datadir)/cowsay/site-cows
+	rm -f $(DESTDIR)$(mandir)/man1/cowthink.1
+	$(LN_S) cowsay.1 $(DESTDIR)$(mandir)/man1/cowthink.1
+	$(INSTALL_DIRS) $(DESTDIR)$(cowsdir)
+	$(INSTALL_DATA) $(COW_FILES) $(DESTDIR)$(cowsdir)
+	$(INSTALL_DIRS) $(DESTDIR)$(sitecowsdir)
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/cowsay $(DESTDIR)$(bindir)/cowthink
 	rm -f $(DESTDIR)$(mandir)/man1/cowsay.1 $(DESTDIR)$(mandir)/man1/cowthink.1
-	rm -rf $(DESTDIR)$(datadir)/cows
+	@set -e; for cow in $(COW_FILES); do dcow="$(DESTDIR)$(cowsdir)/$$(basename "$$cow")"; if test -f "$$dcow"; then echo "rm -f $$dcow"; rm -f "$$dcow"; fi; done
+	@set -e; for dir in $(cowsdir) $(sitecowsdir) $(pkgdatadir) $(etccowpathddir) $(pkgsysconfdir); do if test -d "$(DESTDIR)$${dir}"; then echo rmdir "$(DESTDIR)$${dir}"; rmdir "$(DESTDIR)$${dir}"; fi; done
```

Comment:
Okay, one last thing: The `$sitecowsdir` is the place for local/end-user system administrators to drop their local cow definitions. Those site-local cowfile definitions are not part of the package installation; they're "consumer" or site data. So I think the `cowsay` uninstall process should only attempt to remove the `$sitecowsdir` directory if it is empty. Otherwise, with the `set -e` in place, that `rmdir` would fail and abort the rest of the uninstall process, wouldn't it? I think of site-local files kinda like user-specific data or preference files: uninstalling the application shouldn't remove them, but also their presence should not abort the application/package uninstallation process or cause it to fail.

Similar deal for the `$etccowpathddir` dir: That is a place for either local site administrators, or other packages (possibly supplied by a package manager like a Linux distro, or by a site; either way, core cowsay itself does not know their details) to drop configuration files that are part of _that_ package, not the main cowsay package, but which are managed gracefully by the package manager system. So I think `$etccowpathddir` should only be removed here if it is empty (after `cowsay` removes any main-`cowsay`-package-supplied files inside it as part of the main `cowsay` package uninstallation (I can't remember if there actually are any of those files in the main cowsay pacakge)).

Reply:
You are absolutely correct. It must have gotten lost somewhere in there. I meant to make the actual rmdir command to be

    rmdir "$(DESTDIR)$${dir}" || :

so that if rmdir fails (due to e.g. files existing inside the directory), the removal process would just continue and eventually finish successfully. I will fix that shortly.

Reply:
I have just force-pushed this fix.